pkg/porter/publish.go

```diff
@@ -0,0 +1,172 @@
+package porter
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/deislabs/porter/pkg/config"
+	"github.com/docker/cli/cli/command"
+	cliflags "github.com/docker/cli/cli/flags"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/term"
+	"github.com/docker/docker/registry"
+	"github.com/pkg/errors"
+)
+
+// PublishOptions are options that may be specified when publishing a bundle.
+// Porter handles defaulting any missing values.
+type PublishOptions struct {
+	File     string
+	Tag      string
+	CNABFile string
+}
+
+func (p PublishOptions) Validate(porter *Porter) error {
+	if p.File != "" {
+		return nil
```

Comment:
This is skipping the `Tag` validation entirely. I think the logic below around checking if porter.yaml is present should be in the if/else instead of doing a hard return, no?